connector/src/main/scala/com/datastax/spark/connector/datasource/CassandraCatalog.scala

```diff
@@ -0,0 +1,431 @@
+      .map { case arr =>
+        if (arr.length != 1 && arr.length != 2)
+          throw new CassandraCatalogException(s"Unable to parse clustering column ${arr.mkString(".")}, too many components")
+        if (arr.length == 2) { //Ordering Passed
+          val clusteringOrder = Try(ClusteringOrder.valueOf(arr(1).toUpperCase())).getOrElse(throw new CassandraCatalogException(s"Invalid clustering order found in ${arr.mkString(".")}, must be ASC or DESC or blank"))
+          (fromInternal(arr(0)), clusteringOrder)
+        } else { //No Ordering Passed
+          (fromInternal(arr(0)), ClusteringOrder.ASC)
+        }
+      }
+
+    val protocolVersion = connector.withSessionDo(_.getContext.getProtocolVersion)
+
+    val columnToType = schema.fields.map(sparkField =>
+      (fromInternal(sparkField.name), sparkSqlToJavaDriverType(sparkField.dataType, protocolVersion))
+    ).toMap
+
+    val namespace = fromInternal(ident.namespace.head)
+    val table = fromInternal(ident.name())
+
+    val createTableStart: OngoingPartitionKey = SchemaBuilder.createTable(namespace, table)
+
+    val createTableWithPk: CreateTable = partitionKeyNames.foldLeft(createTableStart) { (createTable, pkName) =>
+      val dataType = columnToType.getOrElse(pkName,
+        throw new CassandraCatalogException(s"$pkName was defined as a partition key but it does not exist in the schema ${schema.fieldNames.mkString(",")}"))
+      createTable.withPartitionKey(pkName, dataType).asInstanceOf[OngoingPartitionKey]
+    }.asInstanceOf[CreateTable]
+
+    val createTableWithClustering = clusteringKeyNames.foldLeft(createTableWithPk) { (createTable, ckName) =>
+      val dataType = columnToType.getOrElse(ckName._1,
+        throw new CassandraCatalogException(s"$ckName was defined as a clustering key but it does not exist in the schema ${schema.fieldNames.mkString(",")}"))
+      createTable
+        .withClusteringColumn(ckName._1, dataType)
+        .withClusteringOrder(ckName._1, ckName._2)
+        .asInstanceOf[CreateTable]
+    }
+
+    val normalColumns = schema.fieldNames.map(fromInternal).toSet -- (clusteringKeyNames.map(_._1) ++ partitionKeyNames)
+
+    val createTableWithColumns = normalColumns.foldRight(createTableWithClustering) { (colName, createTable) =>
+      val dataType = columnToType(colName)
+      createTable.withColumn(colName, dataType)
+    }
+
+    val userProperties = tableProps.filter{ case (key, _) => CassandraProperties.contains(key) }
+    val unusedProperties = tableProps -- userProperties.keys
+    logDebug(s"Ignoring non-cassandra properties for table $unusedProperties")
+    val createTableWithProperties = userProperties.foldLeft(createTableWithColumns) {
+      case (createStmt, (key, value)) => createStmt.withOption(key, parseProperty(value)).asInstanceOf[CreateTable]
+    }
+
+    connector.withSessionDo(_.execute(createTableWithProperties.asCql()))
+
+    loadTable(ident)
```

Comment:
The only case where the session wouldn't get the metadata (because all DDL now blocks for the metadata update if you remember some of our timeouts we had to fix before) the only case where this could be an issue is if the session cache fails and a new session has to be made and this new session is made so quickly it doesn't pick up the metadata. I think this is pretty unlikely ... I think

Reply:
I agree this should be quite unlikely.  We could add some notion of a minimal set of retries to completely guard against such a case, right?  I'm not sure it's worth it, though...

Reply:
I think it's fine :/ I don't really think there is a logical scenario where the session expires while simultaneously being active enough to make the table and not update the metadata.

In my head:
Session is abandoned, therefor lots of time has past and the metadata is updated 
Session is not abandoned, it is the same session that did the DDL and we blocked till metadata was updated